datasources/generic/generic.go

```diff
@@ -124,14 +124,22 @@ func (c *Client) walk(ctx context.Context, root, pathInRoot string, itemChan cha
 			}
 		}
 
+		// When importing a single file, fpath will be '.', and root will
+		// be the full path to the file. Make sure we use that getting
+		// the base name.
+		filename := fpath
```

Comment:
I actually wonder if this is something I can patch in the archiver lib. You'd probably never want "." as a caller. Right? Maybe I can make the archiver's FileFS always return the filename so checks like this are unnecessary.

Reply:
That sounds cleaner, indeed, though I have now idea how that'd affect backwards compat there.

Reply:
I'm not too worried about that since, fortunately, archiver v4 is in alpha state :)

Reply:
Ok, actually, this isn't archiver's doing. It's `io/fs`:

> The path argument contains the argument to [WalkDir](https://pkg.go.dev/io/fs#WalkDir) as a prefix.

And passing in `"."` like we do above is the example in the WalkDir documentation. :thinking: 

So I guess that is _the way_ to walk an entire FS, and I'm not really sure what else to do about it, since that value doesn't flow through the archiver package. Maybe this check will just have to remain :confused:

Anyway, I'm good with merging this PR now if you are.

Reply:
Sounds good to me. The check is a bit weird, but there are some tests in place and it's simple enough. The alternative to treat non-archive regular files is considerably more complex.